reactor-core/src/main/java/reactor/core/publisher/FluxDoOnEach.java

```diff
@@ -100,13 +115,19 @@ public void request(long n) {
 		@Override
 		public void cancel() {
 			s.cancel();
+			if (onSubscriptionSignal != null) {
+				onSubscriptionSignal.accept(SignalType.CANCEL, cachedContext);
```

Comment:
Noob question: what happens if the callback throws an exception, and should we care about it?

Reply:
not a noob question at all, it should indeed be protected.